app/code/community/HE/TwoFactorAuth/controllers/Adminhtml/TwofactorController.php

```diff
@@ -207,13 +219,23 @@ private function _verifyGoogle()
                         );
                     }
 
-                    $msg = Mage::helper('he_twofactorauth')->__("Valid code entered");
-                    Mage::getSingleton('adminhtml/session')->addSuccess($msg);
                     Mage::getSingleton('admin/session')->set2faState(HE_TwoFactorAuth_Model_Validate::TFA_STATE_ACTIVE);
                     $user = Mage::getSingleton('admin/session')->getUser();
                     $password = $user->getPassword();
-                    Mage::dispatchEvent('twofactor_auth_verification_success', array('password' => $password, 'user' => $user, 'result' => true));
-                    $this->_redirect('*');
+                    $lockInfo = new Varien_Object();
+                    Mage::dispatchEvent('twofactor_auth_verification_success', array('password' => $password, 'user' => $user, 'result' => true, 'lock_info' => $lockInfo));
+                    if ($lockInfo->getData('is_locked')) {
+                        Mage::getSingleton('adminhtml/session')->addError(Mage::helper('adminhtml')->__('You did not sign in correctly or your account is temporarily disabled.'));
```

Comment:
This messaging is to be consistent with SUPEE 9767 v2. 

We are deliberately hiding if the user is locked (or) this is a wrong password